File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int letters(string t);
int words(string t);
int sentences(string t);
float l, w, s, a_l, a_s, index;
int a;

int main(void)
{
    //get text from user
    string t = get_string("Text: ");

    //calculate number of letters
    l = letters(t);

    //calculate number of words
    w = words(t);

    //calculate number of sentences
    s = sentences(t);

    //calculate average letters per 100 words
    a_l = (l / w) * 100;

    //calculate average sentences per 100 words
    a_s = 100 / (w / s);

    //plug averages into formula
    index = .0588 * a_l - .296 * a_s - 15.8;
    a = round(index);
    if (a >= 1 && a <= 16)
    {
        printf("Grade %i\n", a);
    }
    else if (a < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}


int letters(string t)
{
    int n = strlen(t);
    for (int i = 0; i < n; i++)
    {
        if ((t[i] >= 'a' && t[i] <= 'z') || (t[i] >= 'A' && t[i] <= 'Z'))
        {
            l = l + 1;
        }
    }
    return l;
}

int words(string t)
{
    int n = strlen(t);
    for (int i = 0; i < n; i++)
    {
        if (t[i] == 32)
        {
            w = w + 1;
        }
    }
    return w + 1;
}

int sentences(string t)
{
    int n = strlen(t);
    for (int i = 0; i < n; i++)
    {
        if (t[i] == 33 || t[i] == 46 || t[i] == 63)
        {
            s = s + 1;
        }
    }
    return s;
}

File: sentimental-readability/readability.py
# get text
text = input("Text: ")

# variables total letters, total words, total sentences
lttr = 0
wrds = 1
snt = 0

# fill variables
for element in text:
    if (element >= 'a' and element <= 'z') or (element >= 'A' and element <= 'Z'):
        lttr += 1
    elif element == " ":
        wrds += 1
    elif element == "." or element == "?" or element == "!":
        snt += 1


# calc average number of letters per 100 words
l = (lttr/wrds) * 100

# calc average number of sentences per 100 words
s = 100 / (wrds/snt)

index = round(0.0588 * l - 0.296 * s - 15.8)
print(lttr, wrds, snt)
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")